src/main/java/seedu/address/model/project/EventList.java

```diff
@@ -28,6 +28,17 @@
         this.events.addAll(events);
     }
 
+    /**
+     * Adds an event to this {@code EventList} and return that new {@code EventList}
+     *
+     * @param event {@code Event} to add.
+     */
+    public EventList addEvent(Event event) {
+        List<Event> events = this.events;
+        events.add(event);
+        return new EventList(events);
```

Comment:
Is there a need to return new EventList? Would it be better to return this instead. If immutability is desirable we can change it to only be able to access an unmodifiable event list.

Reply:
resolved